Failure with scikit-image 0.26.0rc0.dev0+git20251025

From https://github.com/hyperspy/hyperspy-extensions-list/actions/runs/18989738677/job/54250172506, the pyxem test suite fails with scikit-image dev.

From a quick look at the scikit-image repository, this may be related to https://github.com/scikit-image/scikit-image/pull/7789, which has been merged mid-July but the failure is only occurring now because nightly development build were updated only recently.

There is an issue with the deprecation in scikit-image, because it should still work, but it would be good to update the code in pyxem anyway.
I may report it to scikit-image if/when I have time for this - I am not familiar with `EllipseModel` and I would need to figure out a minimum reproduce example for the bug report.

```python
=========================== short test summary info ============================
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_simple - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_xf[40-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_xf[46-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_xf[34-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_yf[20-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_yf[10-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_yf[-9-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_rf_lim[2-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_rf_lim[4-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_rf_lim[3.5-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_min[4-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_min[7-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_min[6.5-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_max[3-6.5-6.6-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_max[3-6.5-4-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_max[3-6.5-9-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_max[6.5-3-4-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_ratio_lim[2.1-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_ratio_lim[1.5-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_semi_len_ratio_lim[3-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[0] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[0.5] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[0.9] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[1.3] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[3.141592653589793] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[6.283185307179586] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_different_rotations[9.9] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[5-3-1-2-7-2-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[3-3-1-2-7-2-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[5-5-1-2-7-2-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[5-2-3-2-7-2-True] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[5-2-3-5-7-2-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[5-2-3-5-6-2-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[5-3-1-3-7-1.5-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestIsEllipseGood::test_many_different_ones[1-9-1-5-6-1.1-False] - AttributeError: property 'params' of 'EllipseModel' object has no setter
FAILED tests/utils/test_ransac_ellipse_tools.py::TestGetEllipseModelRansac::test_rf_lim - ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
= 36 failed, 1290 passed, 76 skipped, 3 deselected, 6 xfailed, 6750 warnings in 346.84s (0:05:46) =
```

<details>

<summary>Failures</summary>

```python
=================================== FAILURES ===================================
________________________ TestIsEllipseGood.test_simple _________________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e549010>

    def test_simple(self):
        params = ret._make_ellipse_model_params_focus(30, 50, 30, 20, 0)
        model = ret.EllipseModel()
>       model.params = params
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:37: AttributeError
______________________ TestIsEllipseGood.test_xf[40-True] ______________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e548e60>
xf = 40, out = True

    @mark.parametrize("xf,out", [(40, True), (46, False), (34, False)])
    def test_xf(self, xf, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(40, 50, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:46: AttributeError
_____________________ TestIsEllipseGood.test_xf[46-False] ______________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e548a70>
xf = 46, out = False

    @mark.parametrize("xf,out", [(40, True), (46, False), (34, False)])
    def test_xf(self, xf, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(40, 50, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:46: AttributeError
_____________________ TestIsEllipseGood.test_xf[34-False] ______________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5488c0>
xf = 34, out = False

    @mark.parametrize("xf,out", [(40, True), (46, False), (34, False)])
    def test_xf(self, xf, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(40, 50, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:46: AttributeError
_____________________ TestIsEllipseGood.test_yf[20-False] ______________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5487d0>
yf = 20, out = False

    @mark.parametrize("yf,out", [(20, False), (10, True), (-9, False)])
    def test_yf(self, yf, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(70, 10, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:55: AttributeError
______________________ TestIsEllipseGood.test_yf[10-True] ______________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e548590>
yf = 10, out = True

    @mark.parametrize("yf,out", [(20, False), (10, True), (-9, False)])
    def test_yf(self, yf, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(70, 10, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:55: AttributeError
_____________________ TestIsEllipseGood.test_yf[-9-False] ______________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e548680>
yf = -9, out = False

    @mark.parametrize("yf,out", [(20, False), (10, True), (-9, False)])
    def test_yf(self, yf, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(70, 10, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:55: AttributeError
____________________ TestIsEllipseGood.test_rf_lim[2-False] ____________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e548230>
rf_lim = 2, out = False

    @mark.parametrize("rf_lim,out", [(2, False), (4, True), (3.5, True)])
    def test_rf_lim(self, rf_lim, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(47, 30, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:64: AttributeError
____________________ TestIsEllipseGood.test_rf_lim[4-True] _____________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e548110>
rf_lim = 4, out = True

    @mark.parametrize("rf_lim,out", [(2, False), (4, True), (3.5, True)])
    def test_rf_lim(self, rf_lim, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(47, 30, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:64: AttributeError
___________________ TestIsEllipseGood.test_rf_lim[3.5-True] ____________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5a3e00>
rf_lim = 3.5, out = True

    @mark.parametrize("rf_lim,out", [(2, False), (4, True), (3.5, True)])
    def test_rf_lim(self, rf_lim, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(47, 30, 10, 30, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:64: AttributeError
_________________ TestIsEllipseGood.test_semi_len_min[4-True] __________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5a36b0>
semi_len_min = 4, out = True

    @mark.parametrize("semi_len_min,out", [(4, True), (7, False), (6.5, True)])
    def test_semi_len_min(self, semi_len_min, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 6.6, 10, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:73: AttributeError
_________________ TestIsEllipseGood.test_semi_len_min[7-False] _________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5a3ec0>
semi_len_min = 7, out = False

    @mark.parametrize("semi_len_min,out", [(4, True), (7, False), (6.5, True)])
    def test_semi_len_min(self, semi_len_min, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 6.6, 10, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:73: AttributeError
________________ TestIsEllipseGood.test_semi_len_min[6.5-True] _________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5a3dd0>
semi_len_min = 6.5, out = True

    @mark.parametrize("semi_len_min,out", [(4, True), (7, False), (6.5, True)])
    def test_semi_len_min(self, semi_len_min, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 6.6, 10, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:73: AttributeError
_____________ TestIsEllipseGood.test_semi_len_max[3-6.5-6.6-True] ______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972dedaae0>
a = 3, b = 6.5, semi_len_max = 6.6, out = True

    @mark.parametrize(
        "a,b,semi_len_max,out",
        [
            (3, 6.5, 6.6, True),
            (3, 6.5, 4, False),
            (3, 6.5, 9, True),
            (6.5, 3, 4, False),
        ],
    )
    def test_semi_len_max(self, a, b, semi_len_max, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, a, b, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:95: AttributeError
______________ TestIsEllipseGood.test_semi_len_max[3-6.5-4-False] ______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972def3650>
a = 3, b = 6.5, semi_len_max = 4, out = False

    @mark.parametrize(
        "a,b,semi_len_max,out",
        [
            (3, 6.5, 6.6, True),
            (3, 6.5, 4, False),
            (3, 6.5, 9, True),
            (6.5, 3, 4, False),
        ],
    )
    def test_semi_len_max(self, a, b, semi_len_max, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, a, b, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:95: AttributeError
______________ TestIsEllipseGood.test_semi_len_max[3-6.5-9-True] _______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df50200>
a = 3, b = 6.5, semi_len_max = 9, out = True

    @mark.parametrize(
        "a,b,semi_len_max,out",
        [
            (3, 6.5, 6.6, True),
            (3, 6.5, 4, False),
            (3, 6.5, 9, True),
            (6.5, 3, 4, False),
        ],
    )
    def test_semi_len_max(self, a, b, semi_len_max, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, a, b, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:95: AttributeError
______________ TestIsEllipseGood.test_semi_len_max[6.5-3-4-False] ______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df50c50>
a = 6.5, b = 3, semi_len_max = 4, out = False

    @mark.parametrize(
        "a,b,semi_len_max,out",
        [
            (3, 6.5, 6.6, True),
            (3, 6.5, 4, False),
            (3, 6.5, 9, True),
            (6.5, 3, 4, False),
        ],
    )
    def test_semi_len_max(self, a, b, semi_len_max, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, a, b, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:95: AttributeError
_____________ TestIsEllipseGood.test_semi_len_ratio_lim[2.1-True] ______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52f00>
semi_len_ratio_lim = 2.1, out = True

    @mark.parametrize("semi_len_ratio_lim,out", [(2.1, True), (1.5, False), (3, True)])
    def test_semi_len_ratio_lim(self, semi_len_ratio_lim, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 9, 4.5, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:115: AttributeError
_____________ TestIsEllipseGood.test_semi_len_ratio_lim[1.5-False] _____________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52d20>
semi_len_ratio_lim = 1.5, out = False

    @mark.parametrize("semi_len_ratio_lim,out", [(2.1, True), (1.5, False), (3, True)])
    def test_semi_len_ratio_lim(self, semi_len_ratio_lim, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 9, 4.5, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:115: AttributeError
______________ TestIsEllipseGood.test_semi_len_ratio_lim[3-True] _______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e5a3830>
semi_len_ratio_lim = 3, out = True

    @mark.parametrize("semi_len_ratio_lim,out", [(2.1, True), (1.5, False), (3, True)])
    def test_semi_len_ratio_lim(self, semi_len_ratio_lim, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 9, 4.5, 0)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:115: AttributeError
________________ TestIsEllipseGood.test_different_rotations[0] _________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972e549670>
r = 0

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
_______________ TestIsEllipseGood.test_different_rotations[0.5] ________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52a80>
r = 0.5

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
_______________ TestIsEllipseGood.test_different_rotations[0.9] ________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52750>
r = 0.9

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
_______________ TestIsEllipseGood.test_different_rotations[1.3] ________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52810>
r = 1.3

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
________ TestIsEllipseGood.test_different_rotations[3.141592653589793] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df528a0>
r = 3.141592653589793

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
________ TestIsEllipseGood.test_different_rotations[6.283185307179586] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52930>
r = 6.283185307179586

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
_______________ TestIsEllipseGood.test_different_rotations[9.9] ________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52480>
r = 9.9

    @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
    def test_different_rotations(self, r):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:129: AttributeError
_________ TestIsEllipseGood.test_many_different_ones[5-3-1-2-7-2-True] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df521e0>
x = 5, y = 3, r = 1, smin = 2, smax = 7, srl = 2, out = True

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
________ TestIsEllipseGood.test_many_different_ones[3-3-1-2-7-2-False] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df52060>
x = 3, y = 3, r = 1, smin = 2, smax = 7, srl = 2, out = False

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
________ TestIsEllipseGood.test_many_different_ones[5-5-1-2-7-2-False] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df51d30>
x = 5, y = 5, r = 1, smin = 2, smax = 7, srl = 2, out = False

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
_________ TestIsEllipseGood.test_many_different_ones[5-2-3-2-7-2-True] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df51d00>
x = 5, y = 2, r = 3, smin = 2, smax = 7, srl = 2, out = True

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
________ TestIsEllipseGood.test_many_different_ones[5-2-3-5-7-2-False] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df51ca0>
x = 5, y = 2, r = 3, smin = 5, smax = 7, srl = 2, out = False

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
________ TestIsEllipseGood.test_many_different_ones[5-2-3-5-6-2-False] _________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df51a30>
x = 5, y = 2, r = 3, smin = 5, smax = 6, srl = 2, out = False

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
_______ TestIsEllipseGood.test_many_different_ones[5-3-1-3-7-1.5-False] ________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df51b20>
x = 5, y = 3, r = 1, smin = 3, smax = 7, srl = 1.5, out = False

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
_______ TestIsEllipseGood.test_many_different_ones[1-9-1-5-6-1.1-False] ________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestIsEllipseGood object at 0x7f972df51bb0>
x = 1, y = 9, r = 1, smin = 5, smax = 6, srl = 1.1, out = False

    @mark.parametrize(
        "x, y, r, smin, smax, srl, out",
        [
            (5, 3, 1, 2, 7, 2, True),
            (3, 3, 1, 2, 7, 2, False),
            (5, 5, 1, 2, 7, 2, False),
            (5, 2, 3, 2, 7, 2, True),
            (5, 2, 3, 5, 7, 2, False),
            (5, 2, 3, 5, 6, 2, False),
            (5, 3, 1, 3, 7, 1.5, False),
            (1, 9, 1, 5, 6, 1.1, False),
        ],
    )
    def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
        model = ret.EllipseModel()
>       model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
        ^^^^^^^^^^^^
E       AttributeError: property 'params' of 'EllipseModel' object has no setter

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:150: AttributeError
____________________ TestGetEllipseModelRansac.test_rf_lim _____________________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestGetEllipseModelRansac object at 0x7f972df81c70>

    def test_rf_lim(self):
        xf, yf = np.ones((2, 3)) * 200, np.ones((2, 3)) * 210
        np.random.seed(7)
        semi0 = np.random.randint(90, 110, size=(2, 3))
        semi1 = np.random.randint(130, 140, size=(2, 3))
        rot = np.zeros((2, 3))
        peak_array = mdtd._make_4d_peak_array_test_data(
            xf, yf, semi0, semi1, rot, nt=20
        )
        ellipse_array0, inlier_array0 = ret.get_ellipse_model_ransac(
            peak_array,
            xf=200,
            yf=210,
            rf_lim=20,
            semi_len_min=80,
            semi_len_max=150,
            semi_len_ratio_lim=1.7,
            min_samples=15,
            max_trials=20,
        )
        ellipse_array1, inlier_array1 = ret.get_ellipse_model_ransac(
            peak_array,
            xf=200,
            yf=210,
            rf_lim=20,
            semi_len_min=80,
            semi_len_max=150,
            semi_len_ratio_lim=1.01,
            min_samples=15,
            max_trials=20,
        )
        for iy, ix in np.ndindex(xf.shape):
            ellipse_params0 = ellipse_array0[iy, ix]
            ellipse_params1 = ellipse_array1[iy, ix]
>           assert ellipse_params0 != ellipse_params1
E           ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()

/home/runner/miniconda3/envs/test/lib/python3.12/site-packages/pyxem/tests/utils/test_ransac_ellipse_tools.py:666: ValueError
```
</details>

On a related topic, I see a random failure regularly: 
https://github.com/hyperspy/hyperspy-bundle/actions/runs/18998570910/job/54261974314?pr=139

```python
================================== FAILURES ===================================
______________ TestGetEllipseModelRansacSingleFrame.test_simple _______________

self = <pyxem.tests.utils.test_ransac_ellipse_tools.TestGetEllipseModelRansacSingleFrame object at 0x000001ECAE23EC60>

    def test_simple(self):
        data = ret.make_ellipse_data_points(50, 55, 20, 16, 2, nt=15)
>       ret.get_ellipse_model_ransac_single_frame(data)

D:\a\hyperspy-bundle\hyperspy-bundle\nd\Lib\site-packages\pyxem\tests\utils\test_ransac_ellipse_tools.py:313: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
D:\a\hyperspy-bundle\hyperspy-bundle\nd\Lib\site-packages\pyxem\utils\ransac_ellipse_tools.py:341: in get_ellipse_model_ransac_single_frame
    model_ransac, inliers = ransac(
D:\a\hyperspy-bundle\hyperspy-bundle\nd\Lib\site-packages\skimage\measure\fit.py:925: in ransac
    success = model.estimate(*samples)
              ^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <skimage.measure.fit.EllipseModel object at 0x000001ECB562FB60>
data = array([[-0.13164413,  1.5087861 ],
       [-0.65890694, -1.1603143 ],
       [ 1.1935532 , -0.84326506],
       [-1.0619094 , -0.6655204 ],
       [-0.64991224,  1.4200842 ],
       [ 1.3088189 , -0.2597694 ]], dtype=float32)

    def estimate(self, data):
        """Estimate ellipse model from data using total least squares.
    
        Parameters
        ----------
        data : (N, 2) array
            N points with ``(x, y)`` coordinates, respectively.
    
        Returns
        -------
        success : bool
            True, if model estimation succeeds.
    
    
        References
        ----------
        .. [1] Halir, R.; Flusser, J. "Numerically stable direct least squares
               fitting of ellipses". In Proc. 6th International Conference in
               Central Europe on Computer Graphics and Visualization.
               WSCG (Vol. 98, pp. 125-132).
    
        """
        # Original Implementation: Ben Hammel, Nick Sullivan-Molina
        # another REFERENCE: [2] http://mathworld.wolfram.com/Ellipse.html
        _check_data_dim(data, dim=2)
    
        if len(data) < 5:
            warn(
                "Need at least 5 data points to estimate an ellipse.",
                category=RuntimeWarning,
                stacklevel=2,
            )
            return False
    
        # to prevent integer overflow, cast data to float, if it isn't already
        float_type = np.promote_types(data.dtype, np.float32)
        data = data.astype(float_type, copy=False)
    
        # normalize value range to avoid misfitting due to numeric errors if
        # the relative distances are small compared to absolute distances
        origin = data.mean(axis=0)
        data = data - origin
        scale = data.std()
        if scale < np.finfo(float_type).tiny:
            warn(
                "Standard deviation of data is too small to estimate "
                "ellipse with meaningful precision.",
                category=RuntimeWarning,
                stacklevel=2,
            )
            return False
        data /= scale
    
        x = data[:, 0]
        y = data[:, 1]
    
        # Quadratic part of design matrix [eqn. 15] from [1]
        D1 = np.vstack([x**2, x * y, y**2]).T
        # Linear part of design matrix [eqn. 16] from [1]
        D2 = np.vstack([x, y, np.ones_like(x)]).T
    
        # forming scatter matrix [eqn. 17] from [1]
        S1 = D1.T @ D1
        S2 = D1.T @ D2
        S3 = D2.T @ D2
    
        # Constraint matrix [eqn. 18]
        C1 = np.array([[0.0, 0.0, 2.0], [0.0, -1.0, 0.0], [2.0, 0.0, 0.0]])
    
        try:
            # Reduced scatter matrix [eqn. 29]
            M = inv(C1) @ (S1 - S2 @ inv(S3) @ S2.T)
        except np.linalg.LinAlgError:  # LinAlgError: Singular matrix
            return False
    
        # M*|a b c >=l|a b c >. Find eigenvalues and eigenvectors
        # from this equation [eqn. 28]
        eig_vals, eig_vecs = np.linalg.eig(M)
    
        # eigenvector must meet constraint 4ac - b^2 to be valid.
        cond = 4 * np.multiply(eig_vecs[0, :], eig_vecs[2, :]) - np.power(
            eig_vecs[1, :], 2
        )
        a1 = eig_vecs[:, (cond > 0)]
        # seeks for empty matrix
        if 0 in a1.shape or len(a1.ravel()) != 3:
            return False
        a, b, c = a1.ravel()
    
        # |d f g> = -S3^(-1)*S2^(T)*|a b c> [eqn. 24]
        a2 = -inv(S3) @ S2.T @ a1
        d, f, g = a2.ravel()
    
        # eigenvectors are the coefficients of an ellipse in general form
        # a*x^2 + 2*b*x*y + c*y^2 + 2*d*x + 2*f*y + g = 0 (eqn. 15) from [2]
        b /= 2.0
        d /= 2.0
        f /= 2.0
    
        # finding center of ellipse [eqn.19 and 20] from [2]
        x0 = (c * d - b * f) / (b**2.0 - a * c)
        y0 = (a * f - b * d) / (b**2.0 - a * c)
    
        # Find the semi-axes lengths [eqn. 21 and 22] from [2]
        numerator = a * f**2 + c * d**2 + g * b**2 - 2 * b * d * f - a * c * g
        term = np.sqrt((a - c) ** 2 + 4 * b**2)
        denominator1 = (b**2 - a * c) * (term - (a + c))
        denominator2 = (b**2 - a * c) * (-term - (a + c))
        width = np.sqrt(2 * numerator / denominator1)
        height = np.sqrt(2 * numerator / denominator2)
    
        # angle of counterclockwise rotation of major-axis of ellipse
        # to x-axis [eqn. 23] from [2].
        phi = 0.5 * np.arctan((2.0 * b) / (a - c))
        if a > c:
            phi += 0.5 * np.pi
    
        # stabilize parameters:
        # sometimes small fluctuations in data can cause
        # height and width to swap
        if width < height:
            width, height = height, width
            phi += np.pi / 2
    
>       phi %= np.pi
E       TypeError: unsupported operand type(s) for %=: 'numpy.complex128' and 'float'

D:\a\hyperspy-bundle\hyperspy-bundle\nd\Lib\site-packages\skimage\measure\fit.py:541: TypeError
```

--- a/pyxem/tests/dummy_data/test_dummy_data.py
+++ b/pyxem/tests/dummy_data/test_dummy_data.py
@@ -17,6 +17,8 @@
 # along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
 
 import pytest
+import skimage
+from packaging.version import Version
 import pyxem.data.dummy_data as dd
 
 
@@ -100,6 +102,10 @@ class TestDummyDataModule:
         assert s.axes_manager.shape == (2, 3, 20, 25)
         assert hasattr(s, "plot")
 
+    @pytest.mark.skipif(
+        Version(skimage.__version__) < Version("0.26.0"),
+        reason="scikit-image >=0.26.0 is required for this functionality.",
+    )
     def test_get_simple_ellipse_signal_peak_array(self):
         s, peak_array = dd.get_simple_ellipse_signal_peak_array()
         s.add_peak_array_as_markers(peak_array=peak_array)

--- a/pyxem/tests/dummy_data/test_make_diffraction_test_data.py
+++ b/pyxem/tests/dummy_data/test_make_diffraction_test_data.py
@@ -20,6 +20,8 @@
 import pytest
 from pytest import approx
 import numpy as np
+import skimage
+from packaging.version import Version
 from hyperspy.signals import Signal2D
 from pyxem.data.dummy_data import make_diffraction_test_data as mdtd
 
@@ -754,6 +756,10 @@ class TestGetEllipticalMask:
         assert not ellipse_image.any()
 
 
+@pytest.mark.skipif(
+    Version(skimage.__version__) < Version("0.26.0"),
+    reason="scikit-image >=0.26.0 is required for this functionality.",
+)
 class TestMake4dPeakArrayTestData:
     def test_simple(self):
         xf, yf = np.ones((2, 3)), np.ones((2, 3))

--- a/pyxem/tests/signals/test_pixelated_stem_class.py
+++ b/pyxem/tests/signals/test_pixelated_stem_class.py
@@ -30,6 +30,8 @@ from pyxem.signals import Diffraction2D, LazyDiffraction2D
 import pyxem.utils.ransac_ellipse_tools as ret
 from pyxem.data.dummy_data import make_diffraction_test_data as mdtd
 import pyxem.data.dummy_data as dd
+import skimage
+from packaging.version import Version
 
 
 @pytest.mark.slow
@@ -166,6 +168,10 @@ class TestAddPeakArrayAsMarkers:
         assert marker.kwargs["offsets"].dtype != object
 
 
+@pytest.mark.skipif(
+    Version(skimage.__version__) < Version("0.26.0"),
+    reason="scikit-image >=0.26.0 is required for this functionality.",
+)
 class TestAddEllipseArrayAsMarkers:
     def test_simple(self):
         s, parray = dd.get_simple_ellipse_signal_peak_array(seed=15)

--- a/pyxem/tests/utils/test_ransac_ellipse_tools.py
+++ b/pyxem/tests/utils/test_ransac_ellipse_tools.py
@@ -20,7 +20,7 @@ from pytest import approx, mark
 import math
 
 import numpy as np
-from numpy.testing import assert_allclose
+import pytest
 from scipy.signal import convolve2d
 from skimage.measure import EllipseModel
 from skimage import morphology
@@ -29,6 +29,9 @@ from hyperspy.signals import Signal2D
 from pyxem.data.dummy_data import make_diffraction_test_data as mdtd
 from pyxem.signals import Diffraction2D
 import pyxem.utils.ransac_ellipse_tools as ret
+
+
+pytest.importorskip("skimage", minversion="0.26", reason="API changes in EllipseModel")
 
 
 class TestIsEllipseGood:

--- a/pyxem/tests/utils/test_ransac_ellipse_tools.py
+++ b/pyxem/tests/utils/test_ransac_ellipse_tools.py
@@ -22,6 +22,7 @@ import math
 import numpy as np
 from numpy.testing import assert_allclose
 from scipy.signal import convolve2d
+from skimage.measure import EllipseModel
 from skimage import morphology
 from hyperspy.signals import Signal2D
 
@@ -33,8 +34,7 @@ import pyxem.utils.ransac_ellipse_tools as ret
 class TestIsEllipseGood:
     def test_simple(self):
         params = ret._make_ellipse_model_params_focus(30, 50, 30, 20, 0)
-        model = ret.EllipseModel()
-        model.params = params
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model, data=None, xf=30, yf=50, rf_lim=5
         )
@@ -42,8 +42,8 @@ class TestIsEllipseGood:
 
     @mark.parametrize("xf,out", [(40, True), (46, False), (34, False)])
     def test_xf(self, xf, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(40, 50, 10, 30, 0)
+        params = ret._make_ellipse_model_params_focus(40, 50, 10, 30, 0)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model, data=None, xf=xf, yf=50, rf_lim=5
         )
@@ -51,8 +51,8 @@ class TestIsEllipseGood:
 
     @mark.parametrize("yf,out", [(20, False), (10, True), (-9, False)])
     def test_yf(self, yf, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(70, 10, 10, 30, 0)
+        params = ret._make_ellipse_model_params_focus(70, 10, 10, 30, 0)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model, data=None, xf=70, yf=yf, rf_lim=5
         )
@@ -60,8 +60,8 @@ class TestIsEllipseGood:
 
     @mark.parametrize("rf_lim,out", [(2, False), (4, True), (3.5, True)])
     def test_rf_lim(self, rf_lim, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(47, 30, 10, 30, 0)
+        params = ret._make_ellipse_model_params_focus(47, 30, 10, 30, 0)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model, data=None, xf=50, yf=30, rf_lim=rf_lim
         )
@@ -69,8 +69,8 @@ class TestIsEllipseGood:
 
     @mark.parametrize("semi_len_min,out", [(4, True), (7, False), (6.5, True)])
     def test_semi_len_min(self, semi_len_min, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(50, 30, 6.6, 10, 0)
+        params = ret._make_ellipse_model_params_focus(50, 30, 6.6, 10, 0)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model,
             data=None,
@@ -91,14 +91,8 @@ class TestIsEllipseGood:
         ],
     )
     def test_semi_len_max(self, a, b, semi_len_max, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(50, 30, a, b, 0)
-        params = list(model.params)
-        # Changing the params like this is necessary to test if b is larger
-        # than semi_len_max. This is due to the content in model.params not being
-        # the same as the input in ret._make_ellipse_model_params_focus.
-        params[2:5] = a, b, 0.0
-        model.params = tuple(params)
+        params = ret._make_ellipse_model_params_focus(50, 30, a, b, 0)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model,
             data=None,
@@ -111,8 +105,8 @@ class TestIsEllipseGood:
 
     @mark.parametrize("semi_len_ratio_lim,out", [(2.1, True), (1.5, False), (3, True)])
     def test_semi_len_ratio_lim(self, semi_len_ratio_lim, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(50, 30, 9, 4.5, 0)
+        params = ret._make_ellipse_model_params_focus(50, 30, 9, 4.5, 0)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model,
             data=None,
@@ -125,8 +119,8 @@ class TestIsEllipseGood:
 
     @mark.parametrize("r", [0, 0.5, 0.9, 1.3, np.pi, 2 * np.pi, 9.9])
     def test_different_rotations(self, r):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
+        params = ret._make_ellipse_model_params_focus(50, 30, 3, 6.5, r)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model, data=None, xf=50, yf=30, rf_lim=5
         )
@@ -146,8 +140,8 @@ class TestIsEllipseGood:
         ],
     )
     def test_many_different_ones(self, x, y, r, smin, smax, srl, out):
-        model = ret.EllipseModel()
-        model.params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
+        params = ret._make_ellipse_model_params_focus(5, 3, 4, 6.5, 0.1)
+        model = EllipseModel(*params)
         is_good = ret.is_ellipse_good(
             ellipse_model=model,
             data=None,
@@ -291,8 +285,8 @@ class TestEllipseCentreToFocus:
 
 
 def compare_model_params(params0, params1, rel=None, abs=None):
-    xf0, yf0, a0, b0, r0 = params0
-    xf1, yf1, a1, b1, r1 = params1
+    (xf0, yf0), (a0, b0), r0 = params0
+    (xf1, yf1), (a1, b1), r1 = params1
     # only case in the params
     if a1 < b1:
         r1 += math.pi / 2
@@ -330,7 +324,7 @@ class TestGetEllipseModelRansacSingleFrame:
         )
         assert inliers0[:-1].all()
         assert not inliers0[-1]
-        model_ransac1, inliers1 = ret.get_ellipse_model_ransac_single_frame(
+        _, inliers1 = ret.get_ellipse_model_ransac_single_frame(
             data,
             xf=50,
             yf=55,
@@ -364,8 +358,9 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=100,
         )
-        params0f = (xf, yf, a, b, r)
-        params1f = ret._ellipse_model_centre_to_focus(*model_ransac.params, xf, yf)
+        params0f = (xf, yf), (a, b), r
+        params = model_ransac.center, model_ransac.axis_lengths, model_ransac.theta
+        params1f = ret._ellipse_model_centre_to_focus(*params, xf, yf)
         assert inliers.all()
         compare_model_params(params0f, params1f, abs=0.01)
 
@@ -375,7 +370,7 @@ class TestGetEllipseModelRansacSingleFrame:
         data0 = ret.make_ellipse_data_points(xf0, yf, a, b, r, nt=25)
         data1 = ret.make_ellipse_data_points(xf1, yf, a, b, r, nt=25)
         data = np.vstack((data0, data1))
-        model_ransac0, inliers0 = ret.get_ellipse_model_ransac_single_frame(
+        model_ransac0, _ = ret.get_ellipse_model_ransac_single_frame(
             data,
             xf=xf0,
             yf=55,
@@ -387,11 +382,12 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=300,
         )
-        params00f = (xf0, yf, a, b, r)
-        params01f = ret._ellipse_model_centre_to_focus(*model_ransac0.params, xf0, yf)
+        params00f = (xf0, yf), (a, b), r
+        params = model_ransac0.center, model_ransac0.axis_lengths, model_ransac0.theta
+        params01f = ret._ellipse_model_centre_to_focus(*params, xf0, yf)
         compare_model_params(params00f, params01f, abs=0.1)
 
-        model_ransac1, inliers1 = ret.get_ellipse_model_ransac_single_frame(
+        model_ransac1, _ = ret.get_ellipse_model_ransac_single_frame(
             data,
             xf=xf1,
             yf=55,
@@ -403,8 +399,9 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=300,
         )
-        params10f = (xf1, yf, a, b, r)
-        params11f = ret._ellipse_model_centre_to_focus(*model_ransac1.params, xf1, yf)
+        params10f = (xf1, yf), (a, b), r
+        params = model_ransac1.center, model_ransac1.axis_lengths, model_ransac1.theta
+        params11f = ret._ellipse_model_centre_to_focus(*params, xf1, yf)
         compare_model_params(params10f, params11f, abs=0.1)
 
     @mark.skip(reason="Broken due to changes in skimage 0.19")
@@ -425,8 +422,9 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=300,
         )
-        params00f = (xf, yf0, a, b, r)
-        params01f = ret._ellipse_model_centre_to_focus(*model_ransac0.params, xf, yf0)
+        params00f = (xf, yf0), (a, b), r
+        params = model_ransac0.center, model_ransac0.axis_lengths, model_ransac0.theta
+        params01f = ret._ellipse_model_centre_to_focus(*params, xf, yf0)
         compare_model_params(params00f, params01f, abs=0.1)
 
         model_ransac1, inliers1 = ret.get_ellipse_model_ransac_single_frame(
@@ -441,8 +439,9 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=300,
         )
-        params10f = (xf, yf1, a, b, r)
-        params11f = ret._ellipse_model_centre_to_focus(*model_ransac1.params, xf, yf1)
+        params10f = (xf, yf1), (a, b), r
+        params = model_ransac1.center, model_ransac1.axis_lengths, model_ransac1.theta
+        params11f = ret._ellipse_model_centre_to_focus(*params, xf, yf1)
         compare_model_params(params10f, params11f, abs=0.1)
 
     def test_rf_lim(self):
@@ -460,12 +459,13 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=100,
         )
-        params0f = (xf, yf, a, b, r)
-        params1f = ret._ellipse_model_centre_to_focus(*model_ransac0.params, xf, yf)
+        params0f = (xf, yf), (a, b), r
+        params = *model_ransac0.center, *model_ransac0.axis_lengths, model_ransac0.theta
+        params1f = ret._ellipse_model_centre_to_focus(*params, xf, yf)
         assert inliers0.all()
         compare_model_params(params0f, params1f, abs=0.1)
 
-        model_ransac1, inliers1 = ret.get_ellipse_model_ransac_single_frame(
+        model_ransac1, _ = ret.get_ellipse_model_ransac_single_frame(
             data,
             xf=50,
             yf=50,
@@ -494,12 +494,13 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=100,
         )
-        params0f = (xf, yf, a, b, r)
-        params1f = ret._ellipse_model_centre_to_focus(*model_ransac0.params, xf, yf)
+        params0f = (xf, yf), (a, b), r
+        params = *model_ransac0.center, *model_ransac0.axis_lengths, model_ransac0.theta
+        params1f = ret._ellipse_model_centre_to_focus(*params, xf, yf)
         assert inliers0.all()
         compare_model_params(params0f, params1f, abs=0.1)
 
-        model_ransac1, inliers1 = ret.get_ellipse_model_ransac_single_frame(
+        model_ransac1, _ = ret.get_ellipse_model_ransac_single_frame(
             data,
             xf=50,
             yf=55,
@@ -528,12 +529,13 @@ class TestGetEllipseModelRansacSingleFrame:
             residual_threshold=10,
             max_trials=100,
         )
-        params0f = (xf, yf, a, b, r)
-        params1f = ret._ellipse_model_centre_to_focus(*model_ransac0.params, xf, yf)
+        params0f = (xf, yf), (a, b), r
+        params = *model_ransac0.center, *model_ransac0.axis_lengths, model_ransac0.theta
+        params1f = ret._ellipse_model_centre_to_focus(*params, xf, yf)
         assert inliers0.all()
         compare_model_params(params0f, params1f, abs=0.1)
 
-        model_ransac1, inliers1 = ret.get_ellipse_model_ransac_single_frame(
+        model_ransac1, _ = ret.get_ellipse_model_ransac_single_frame(
             data,
             xf=50,
             yf=55,
@@ -829,12 +831,11 @@ def test_full_ellipse_ransac_processing():
         ycf, xcf, bf, af, rf = ellipse_array[iy, ix]
         # shifted by pi/2
         compare_model_params(
-            (xcf, ycf, af, bf, rf + np.pi / 2), (yc, xc, a, b, r), abs=0.1
+            ((xcf, ycf), (af, bf), rf + np.pi / 2), ((yc, xc), (a, b), r), abs=0.1
         )
         assert inlier_array[iy, ix].all()
 
     s.add_ellipse_array_as_markers(ellipse_array)
-    x_list, y_list = [], []
 
 
 class TestDetermineEllipse:

--- a/pyxem/utils/ransac_ellipse_tools.py
+++ b/pyxem/utils/ransac_ellipse_tools.py
@@ -21,12 +21,16 @@
 from tqdm import tqdm
 import math
 from functools import partial
+from packaging.version import Version
+
 import numpy as np
+import skimage
 from skimage.measure import EllipseModel, ransac
 import warnings
 from hyperspy.signals import BaseSignal
 from hyperspy.misc.utils import isiterable
 import hyperspy.api as hs
+
 
 __all__ = [
     "is_ellipse_good",
@@ -38,6 +42,20 @@ __all__ = [
 ]
 
 
+def _requires_scikit_image_026(func):
+    """Decorator that checks if scikit-image >= 0.26.0 is installed."""
+
+    def wrapper(*args, **kwargs):
+        if Version(skimage.__version__) < Version("0.26.0"):
+            raise ImportError(
+                "scikit-image >=0.26.0 is required for this functionality."
+            )
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+@_requires_scikit_image_026
 def is_ellipse_good(
     ellipse_model,
     data,
@@ -170,6 +188,7 @@ def _get_closest_focus(x, y, xc, yc, a, b, r):
     return (xf, yf)
 
 
+@_requires_scikit_image_026
 def make_ellipse_data_points(x, y, a, b, r, nt=20, use_focus=True):
     """Get an ellipse position list.
 
@@ -265,6 +284,7 @@ def _make_ellipse_model_params_focus(xf, yf, a, b, r):
     return (xc, yc), (a, b), r
 
 
+@_requires_scikit_image_026
 def get_ellipse_model_ransac_single_frame(
     data,
     xf=128,
@@ -599,6 +619,7 @@ def ellipse_to_markers(ellipse_array, points=None, inlier=None):
         return el
 
 
+@_requires_scikit_image_026
 def determine_ellipse(
     signal=None,
     pos=None,

--- a/pyxem/utils/ransac_ellipse_tools.py
+++ b/pyxem/utils/ransac_ellipse_tools.py
@@ -73,13 +73,15 @@ def is_ellipse_good(
     Examples
     --------
     >>> import pyxem.utils.ransac_ellipse_tools as ret
-    >>> model = ret.EllipseModel()
-    >>> model.params = ret._make_ellipse_model_params_focus(30, 50, 30, 20, 0)
+    >>> params = ret._make_ellipse_model_params_focus(30, 50, 30, 20, 0)
+    >>> model = ret.EllipseModel(*params)
     >>> is_good = ret.is_ellipse_good(
     ...         ellipse_model=model, data=None, xf=30, yf=50, rf_lim=5)
 
     """
-    xc, yc, a, b, r = ellipse_model.params
+    xc, yc = ellipse_model.center
+    a, b = ellipse_model.axis_lengths
+    r = ellipse_model.theta
     x, y = _get_closest_focus(xf, yf, xc, yc, a, b, r)
     rf = math.hypot(x - xf, y - yf)
     if rf > rf_lim:
@@ -102,7 +104,7 @@ def is_ellipse_good(
 
 
 def _ellipse_centre_to_focus(x, y, a, b, r):
-    """Get focus for an ellipse from EllipseModel.params
+    """Get focus for an ellipse from EllipseModel parameters
 
     Parameters
     ----------
@@ -204,9 +206,9 @@ def make_ellipse_data_points(x, y, a, b, r, nt=20, use_focus=True):
     if use_focus:
         params = _make_ellipse_model_params_focus(x, y, a, b, r)
     else:
-        params = (x, y, a, b, r)
+        params = (x, y), (a, b), r
     theta_array = np.arange(0, 2 * np.pi, 2 * np.pi / nt)
-    data = EllipseModel().predict_xy(theta_array, params=params)
+    data = EllipseModel(*params).predict_xy(theta_array)
     return data
 
 
@@ -233,7 +235,7 @@ def _ellipse_model_centre_to_focus(xc, yc, a, b, r, x, y):
 
     """
     xf, yf = _get_closest_focus(x, y, xc, yc, a, b, r)
-    params = (xf, yf, a, b, r)
+    params = (xf, yf), (a, b), r
     return params
 
 
@@ -251,7 +253,7 @@ def _make_ellipse_model_params_focus(xf, yf, a, b, r):
     Returns
     -------
     params : tuple
-        (xc, yc, a, b, r)
+        (xc, yc), (a, b), r
 
     """
     if a < b:
@@ -260,8 +262,7 @@ def _make_ellipse_model_params_focus(xf, yf, a, b, r):
     c = math.sqrt(math.pow(a, 2) - math.pow(b, 2))
     xc = xf - c * math.cos(r)
     yc = yf - c * math.sin(r)
-    params = (xc, yc, a, b, r)
-    return params
+    return (xc, yc), (a, b), r
 
 
 def get_ellipse_model_ransac_single_frame(
@@ -308,14 +309,14 @@ def get_ellipse_model_ransac_single_frame(
     Returns
     -------
     model_ransac, inliers
-        Model data is accessed in model_ransac.params:
-        [x, y, semi_len0, semi_len1, rotation]
+        Model data is accessed in model_ransac.center,
+        model_ransac.axis_lengths, model_ransac.theta.
 
     Examples
     --------
     >>> import pyxem.utils.ransac_ellipse_tools as ret
-    >>> data = ret.EllipseModel().predict_xy(
-    ...        np.arange(0, 2*np.pi, 0.5), params=(128, 130, 50, 60, 0.2))
+    >>> data = ret.EllipseModel((128, 130), (50, 60), 0.2).predict_xy(
+    ...        np.arange(0, 2*np.pi, 0.5))
     >>> ellipse_model, inliers = ret.get_ellipse_model_ransac_single_frame(
     ...        data, xf=128, yf=128, rf_lim=5, semi_len_min=45,
     ...        semi_len_max=65, semi_len_ratio_lim=1.4, max_trials=1000)
@@ -438,7 +439,11 @@ def get_ellipse_model_ransac(
             max_trials=max_trials,
         )
         if ellipse_model is not None:
-            params = ellipse_model.params
+            params = (
+                *ellipse_model.center,
+                *ellipse_model.axis_lengths,
+                ellipse_model.theta,
+            )
         else:
             params = None
         ellipse_array[iy, ix] = params
@@ -674,16 +679,14 @@ def determine_ellipse(
         else:
             el, inlier = get_ellipse_model_ransac_single_frame(pos, **kwargs)
     else:
-        e = EllipseModel()
-        converge = e.estimate(data=pos)
-        el = e
+        el = EllipseModel.from_estimate(data=pos)
     if el is not None:
-        affine = _ellipse_to_affine(el.params[3], el.params[2], el.params[4])
-        center = (el.params[0], el.params[1])
+        affine = _ellipse_to_affine(el.axis_lengths[1], el.axis_lengths[0], el.theta)
+        center = (el.center[0], el.center[1])
         if return_params and use_ransac:
-            return center, affine, el.params, pos, inlier
+            return center, affine, *el.center, *el.axis_lengths, el.theta, pos, inlier
         elif return_params:
-            return center, affine, el.params, pos
+            return center, affine, *el.center, *el.axis_lengths, el.theta, pos
         else:
             return center, affine
     else:  # pragma: no cover

